--- uClibc-0.9.33.3/libc/sysdeps/linux/i386/brk.c.0035~	2015-05-20 12:25:24.000000000 +0200
+++ uClibc-0.9.33.3/libc/sysdeps/linux/i386/brk.c	2015-06-24 19:41:55.113640161 +0200
@@ -22,22 +22,14 @@
 #include <sys/syscall.h>
 
 /* This must be initialized data because commons can't have aliases.  */
-void *__curbrk attribute_hidden = 0;
+void *__curbrk = 0;
 
 int brk(void *addr)
 {
 	void *newbrk;
 
-	/* %ebx is used in PIC code, need to save/restore it manually.
-	 * gcc won't do it for us if we will request it in constraints
-	 */
-	__asm__("pushl	%%ebx\n"
-		"movl	%2, %%ebx\n"
-		"int	$0x80\n"
-		"popl	%%ebx\n"
-		: "=a" (newbrk)
-		: "0" (__NR_brk), "g" (addr)
-	);
+	INTERNAL_SYSCALL_DECL (err);
+	newbrk = (void *) INTERNAL_SYSCALL (brk, err, 1, addr);
 
 	__curbrk = newbrk;
 
@@ -48,4 +40,3 @@ int brk(void *addr)
 
 	return 0;
 }
-libc_hidden_def(brk)
