--- uClibc-0.9.33.3/libc/sysdeps/linux/common/eventfd_read.c.eventfd~	2014-06-23 15:52:12.785122747 +0200
+++ uClibc-0.9.33.3/libc/sysdeps/linux/common/eventfd_read.c	2014-06-23 15:52:12.785122747 +0200
@@ -0,0 +1,27 @@
+/* Copyright (C) 2007-2014 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#include <errno.h>
+#include <unistd.h>
+#include <sys/eventfd.h>
+
+
+int
+eventfd_read (int fd, eventfd_t *value)
+{
+  return read (fd, value, sizeof (eventfd_t)) != sizeof (eventfd_t) ? -1 : 0;
+}
--- uClibc-0.9.33.3/libc/sysdeps/linux/common/eventfd_write.c.eventfd~	2014-06-23 15:52:12.785122747 +0200
+++ uClibc-0.9.33.3/libc/sysdeps/linux/common/eventfd_write.c	2014-06-23 15:52:12.785122747 +0200
@@ -0,0 +1,28 @@
+/* Copyright (C) 2007-2014 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#include <errno.h>
+#include <unistd.h>
+#include <sys/eventfd.h>
+
+
+int
+eventfd_write (int fd, eventfd_t value)
+{
+  return write (fd, &value,
+		sizeof (eventfd_t)) != sizeof (eventfd_t) ? -1 : 0;
+}
--- uClibc-0.9.33.3/libc/sysdeps/linux/common/Makefile.in.eventfd~	2014-06-23 15:52:12.778456016 +0200
+++ uClibc-0.9.33.3/libc/sysdeps/linux/common/Makefile.in	2014-06-23 15:52:12.785122747 +0200
@@ -25,6 +25,8 @@ CSRC-$(UCLIBC_LINUX_SPECIFIC) += \
 	capset.c \
 	dup3.c \
 	eventfd.c \
+	eventfd_read.c \
+	eventfd_write.c \
 	inotify.c \
 	ioperm.c \
 	iopl.c \
--- uClibc-0.9.33.3/libc/sysdeps/linux/common/sys/eventfd.h.eventfd~	2014-06-23 15:52:12.785122747 +0200
+++ uClibc-0.9.33.3/libc/sysdeps/linux/common/sys/eventfd.h	2014-06-23 15:53:16.109065871 +0200
@@ -43,6 +43,12 @@ __BEGIN_DECLS
    value to COUNT.  */
 extern int eventfd (int __count, int __flags) __THROW;
 
+/* Read event counter and possibly wait for events.  */
+extern int eventfd_read (int __fd, eventfd_t *__value);
+
+/* Increment event counter.  */
+extern int eventfd_write (int __fd, eventfd_t __value);
+
 __END_DECLS
 
 #endif /* sys/eventfd.h */
