--- uClibc-0.9.33.3/libc/sysdeps/linux/i386/brk.c.brk~	2015-06-23 12:21:28.321071840 +0200
+++ uClibc-0.9.33.3/libc/sysdeps/linux/i386/brk.c	2015-06-23 12:20:24.438140235 +0200
@@ -21,23 +21,49 @@
 #include <unistd.h>
 #include <sys/syscall.h>
 
+#undef INTERNAL_SYSCALL_DECL
+#define INTERNAL_SYSCALL_DECL(err) do { } while (0)
+
+#  define INTERNAL_SYSCALL(name, err, nr, args...) \
+  ({									      \
+    register unsigned int resultvar;					      \
+    EXTRAVAR_##nr							      \
+    asm volatile (							      \
+    LOADARGS_##nr							      \
+    "movl %1, %%eax\n\t"						      \
+    "call *%%gs:%P2\n\t"						      \
+    RESTOREARGS_##nr							      \
+    : "=a" (resultvar)							      \
+    : "i" (__NR_##name), "i" (offsetof (tcbhead_t, sysinfo))		      \
+      ASMFMT_##nr(args) : "memory", "cc");				      \
+    (int) resultvar; })
+#  define INTERNAL_SYSCALL_NCS(name, err, nr, args...) \
+  ({									      \
+    register unsigned int resultvar;					      \
+    EXTRAVAR_##nr							      \
+    asm volatile (							      \
+    LOADARGS_##nr							      \
+    "call *%%gs:%P2\n\t"						      \
+    RESTOREARGS_##nr							      \
+    : "=a" (resultvar)							      \
+    : "0" (name), "i" (offsetof (tcbhead_t, sysinfo))			      \
+      ASMFMT_##nr(args) : "memory", "cc");				      \
+    (int) resultvar; })
+
 /* This must be initialized data because commons can't have aliases.  */
-void *__curbrk attribute_hidden = 0;
+void *__curbrk = 0;
+
+/* Old braindamage in GCC's crtstuff.c requires this symbol in an attempt
+   to work around different old braindamage in the old Linux ELF dynamic
+   linker.  */
+weak_alias (__curbrk, ___brk_addr)
 
 int brk(void *addr)
 {
 	void *newbrk;
 
-	/* %ebx is used in PIC code, need to save/restore it manually.
-	 * gcc won't do it for us if we will request it in constraints
-	 */
-	__asm__("pushl	%%ebx\n"
-		"movl	%2, %%ebx\n"
-		"int	$0x80\n"
-		"popl	%%ebx\n"
-		: "=a" (newbrk)
-		: "0" (__NR_brk), "g" (addr)
-	);
+	INTERNAL_SYSCALL_DECL (err);
+	newbrk = (void *) INTERNAL_SYSCALL (brk, err, 1, addr);
 
 	__curbrk = newbrk;
 
@@ -48,4 +74,4 @@ int brk(void *addr)
 
 	return 0;
 }
-libc_hidden_def(brk)
+weak_alias (__brk, brk)
