--- uClibc-0.9.33.2/Makerules~	2013-06-21 07:59:12.683644142 +0200
+++ uClibc-0.9.33.2/Makerules	2013-06-21 08:14:35.801847012 +0200
@@ -310,7 +310,7 @@ define link.so
 		-Wl,-soname=$(notdir $@).$(2) \
 		$(CFLAG_-nostdlib) -o $(dir $@)$(1) $(START_FILE-$(notdir $@)) \
 		-Wl,--whole-archive $(firstword $^) -Wl,--no-whole-archive \
-		$(LIBS-$(notdir $@)) $(LIBGCC) $(END_FILE-$(notdir $@))
+		$(LIBS-$(notdir $@)) $(LIBGCC) $(END_FILE-$(notdir $@)) -lgcc_eh
 	$(Q)$(LN) -sf $(1) $@.$(2)
 	$(Q)$(LN) -sf $(1) $@
 endef
