--- uClibc-0.9.33.3/Makerules.bfd_link~	2014-02-18 14:38:18.811206743 +0100
+++ uClibc-0.9.33.3/Makerules	2014-02-18 14:38:01.031206857 +0100
@@ -53,6 +53,9 @@ $(shared_objs) $(ar_objs): | $(sub_heade
 headers-y: $(headers-y)
 	@true
 
+LDFLAG-fuse-ld := -fuse-ld=bfd
+# deliberately not named CFLAG-fuse-ld since unchecked and from user
+
 MAKEFLAGS += --no-print-directory
 SHELL_SET_X := set +x
 define rel_srcdir
@@ -296,7 +299,7 @@ cmd_hcompile.o = $(HOSTCC) $(filter-out
 
 define create-lds
 	$(Q)$(RM) $@.lds
-	$(Q)$(CC) -nostdlib -nostartfiles -shared -Wl,-z,combreloc \
+	$(Q)$(CC) $(LDFLAG-fuse-ld) -nostdlib -nostartfiles -shared -Wl,-z,combreloc \
 	-Wl,-z,relro -Wl,--hash-style=gnu -Wl,-z,defs \
 	-Wl,--verbose 2>&1 | LC_ALL=C \
 	sed -e '/^=========/,/^=========/!d;/^=========/d' \
@@ -306,7 +309,7 @@ endef
 define link.so
 	$(Q)$(RM) $@ $@.$(2) $(dir $@)$(1)
 	@$(disp_ld)
-	$(Q)$(CC) $(LDFLAGS-$(notdir $@)) $(LDFLAGS-y-$(@F)) \
+	$(Q)$(CC) $(LDFLAG-fuse-ld)  $(LDFLAGS-$(notdir $@)) $(LDFLAGS-y-$(@F)) \
 		-Wl,-soname=$(notdir $@).$(2) \
 		$(CFLAG_-nostdlib) -o $(dir $@)$(1) $(START_FILE-$(notdir $@)) \
 		-Wl,--whole-archive $(firstword $^) -Wl,--no-whole-archive \
--- uClibc-0.9.33.3/Rules.mak.bfd_link~	2014-02-18 14:23:44.971212348 +0100
+++ uClibc-0.9.33.3/Rules.mak	2014-02-18 14:23:44.971212348 +0100
@@ -166,9 +166,10 @@ endif
 check_gcc=$(shell \
 	if $(CC) $(1) -S -o /dev/null -xc /dev/null > /dev/null 2>&1; \
 	then echo "$(1)"; else echo "$(2)"; fi)
+# FIXME: filter -fuse-ld=% and pass that, too!!
 check_as=$(shell \
-	if $(CC) -Wa,$(1) -Wa,-Z -c -o /dev/null -xassembler /dev/null > /dev/null 2>&1; \
-	then echo "-Wa,$(1)"; fi)
+	 if $(CC) $(LDFLAG-fuse-ld) -Wl,$(1) $(CFLAG_-nostdlib) -o /dev/null -Wl,-b,binary /dev/null > /dev/null 2>&1; \
+	then echo "$(1)"; fi)
 check_ld=$(shell \
 	if $(LD) $(1) -o /dev/null -b binary /dev/null > /dev/null 2>&1; \
 	then echo "$(1)"; fi)
@@ -222,6 +223,11 @@ endef
 ARFLAGS:=cr
 
 
+# Note: The check for -nostdlib has to be before all calls to check_ld
+$(eval $(call check-gcc-var,-nostdlib))
+LDFLAG-fuse-ld := $(filter -fuse-ld=%,$(EXTRA_UCLIBC_FLAGS))
+# deliberately not named CFLAG-fuse-ld since unchecked and from user
+
 # Flags in OPTIMIZATION are used only for non-debug builds
 
 OPTIMIZATION:=
@@ -647,7 +653,7 @@ endif
 $(eval $(call check-ld-var,--warn-once))
 $(eval $(call check-ld-var,--sort-common))
 $(eval $(call check-ld-var,--discard-all))
-LDFLAGS_NOSTRIP:=$(CPU_LDFLAGS-y) -shared \
+LDFLAGS_NOSTRIP:=$(LDFLAG-fuse-ld) $(CPU_LDFLAGS-y) -shared \
 	-Wl,--warn-common $(CFLAG_-Wl--warn-once) -Wl,-z,combreloc
 # binutils-2.16.1 warns about ignored sections, 2.16.91.0.3 and newer are ok
 #$(eval $(call check-ld-var,--gc-sections))
