--- uClibc-0.9.33.2/ldso/ldso/dl-elf.c.lib64~	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/ldso/ldso/dl-elf.c	2013-01-07 02:09:28.228559563 +0100
@@ -296,10 +296,10 @@ struct elf_resolve *_dl_load_shared_libr
 	   This list must exactly match the list in uClibc/ldso/util/ldd.c */
 	_dl_if_debug_dprint("\tsearching full lib path list\n");
 	tpnt1 = search_for_named_library(libname, rflags,
-					UCLIBC_RUNTIME_PREFIX "lib:"
-					UCLIBC_RUNTIME_PREFIX "usr/lib"
+					UCLIBC_RUNTIME_PREFIX UCLIBC_MULTILIB_DIR ":"
+					UCLIBC_DEVEL_PREFIX UCLIBC_MULTILIB_DIR
 #ifndef __LDSO_CACHE_SUPPORT__
-					":" UCLIBC_RUNTIME_PREFIX "usr/X11R6/lib"
+					":" UCLIBC_RUNTIME_PREFIX "usr/X11R6/" UCLIBC_MULTILIB_DIR
 #endif
 					, rpnt);
 	if (tpnt1 != NULL)
--- uClibc-0.9.33.2/ldso/ldso/Makefile.in.lib64~	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/ldso/ldso/Makefile.in	2013-01-07 02:09:28.228559563 +0100
@@ -13,7 +13,7 @@ CFLAGS-ldso := -DNOT_IN_libc -DIS_IN_rtl
 CFLAGS-ldso += -fno-omit-frame-pointer
 
 CFLAGS-ldso += -I$(top_srcdir)ldso/ldso/$(TARGET_ARCH) -I$(top_srcdir)ldso/include -I$(top_srcdir)ldso/ldso
-CFLAGS-ldso += -DUCLIBC_RUNTIME_PREFIX=\"$(RUNTIME_PREFIX)\" -DUCLIBC_LDSO=\"$(UCLIBC_LDSO)\"
+CFLAGS-ldso += -DUCLIBC_RUNTIME_PREFIX=\"$(RUNTIME_PREFIX)\" -DUCLIBC_DEVEL_PREFIX=\"$(DEVEL_PREFIX)\" -DUCLIBC_MULTILIB_DIR=\"$(MULTILIB_DIR)\" -DUCLIBC_LDSO=\"$(UCLIBC_LDSO)\"
 
 ifeq ($(DODEBUG),y)
 # Not really much point in including debugging info, since gdb
--- uClibc-0.9.33.2/ldso/libdl/Makefile.in.lib64~	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/ldso/libdl/Makefile.in	2013-01-07 02:09:28.228559563 +0100
@@ -11,7 +11,7 @@ CFLAGS-libdl := -DNOT_IN_libc -DIS_IN_li
 
 CFLAGS-libdl += -I$(top_srcdir)ldso/ldso/$(TARGET_ARCH) -I$(top_srcdir)ldso/include -I$(top_srcdir)ldso/ldso
 
-CFLAGS-libdl += -DUCLIBC_RUNTIME_PREFIX=\"$(RUNTIME_PREFIX)\"
+CFLAGS-libdl += -DUCLIBC_RUNTIME_PREFIX=\"$(RUNTIME_PREFIX)\" -DUCLIBC_DEVEL_PREFIX=\"$(DEVEL_PREFIX)\" -DUCLIBC_MULTILIB_DIR=\"$(MULTILIB_DIR)\"
 
 ifeq ($(SUPPORT_LD_DEBUG),y)
 CFLAGS-libdl += -D__SUPPORT_LD_DEBUG__
--- uClibc-0.9.33.2/Rules.mak.lib64~	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/Rules.mak	2013-01-07 02:10:25.486836632 +0100
@@ -96,7 +96,7 @@ export ARCH
 TARGET_SUBARCH:=$(call qstrip,$(TARGET_SUBARCH))
 RUNTIME_PREFIX:=$(strip $(subst //,/, $(subst ,/, $(call qstrip,$(RUNTIME_PREFIX)))))
 DEVEL_PREFIX:=$(strip $(subst //,/, $(subst ,/, $(call qstrip,$(DEVEL_PREFIX)))))
-MULTILIB_DIR:=$(strip $(subst //,/, $(subst ,/, $(call qstrip,$(MULTILIB_DIR)))))
+MULTILIB_DIR:=$(strip $(subst //,/, $(call qstrip,$(MULTILIB_DIR))))
 KERNEL_HEADERS:=$(strip $(subst //,/, $(subst ,/, $(call qstrip,$(KERNEL_HEADERS)))))
 export RUNTIME_PREFIX DEVEL_PREFIX KERNEL_HEADERS MULTILIB_DIR
 
@@ -577,7 +577,7 @@ ifndef ASNEEDED
 export ASNEEDED:=$(shell $(LD) --help 2>/dev/null | grep -q -- --as-needed && echo "AS_NEEDED ( $(UCLIBC_LDSO) )" || echo "$(UCLIBC_LDSO)")
 ifeq ($(UCLIBC_HAS_BACKTRACE),y)
 # Only used in installed libc.so linker script
-UBACKTRACE_FULL_NAME := $(RUNTIME_PREFIX)lib/$(UBACKTRACE_DSO)
+UBACKTRACE_FULL_NAME := $(RUNTIME_PREFIX)$(MULTILIB_DIR)/$(UBACKTRACE_DSO)
 export UBACKTRACE_ASNEEDED:=$(shell $(LD) --help 2>/dev/null | grep -q -- --as-needed && echo "AS_NEEDED ( $(UBACKTRACE_FULL_NAME) )" || echo "$(UBACKTRACE_FULL_NAME)")
 else
 export UBACKTRACE_ASNEEDED:=""
--- uClibc-0.9.33.2/utils/ldconfig.c.lib64~	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/utils/ldconfig.c	2013-01-07 02:09:28.237559765 +0100
@@ -949,10 +949,10 @@ int main(int argc, char **argv)
 
 		/* look ma, no defaults */
 		if (!nodefault) {
-			scan_dir(UCLIBC_RUNTIME_PREFIX "lib");
-			scan_dir(UCLIBC_RUNTIME_PREFIX "usr/lib");
+			scan_dir(UCLIBC_RUNTIME_PREFIX UCLIBC_MULTILIB_DIR);
+			scan_dir(UCLIBC_DEVEL_PREFIX UCLIBC_MULTILIB_DIR);
 #ifndef __LDSO_CACHE_SUPPORT__
-			scan_dir(UCLIBC_RUNTIME_PREFIX "usr/X11R6/lib");
+			scan_dir(UCLIBC_RUNTIME_PREFIX "usr/X11R6/" UCLIBC_MULTILIB_DIR);
 #else
 			/* I guess the defaults aren't good enough */
 			if ((extpath = get_extpath())) {
@@ -963,8 +963,8 @@ int main(int argc, char **argv)
 						while (cp[--len] == '/' && len)
 							cp[len] = 0;
 					/* we do the redundancy check only if cache usage is enabled */
-					if (strcmp(UCLIBC_RUNTIME_PREFIX "lib", cp) == 0
-					    || strcmp(UCLIBC_RUNTIME_PREFIX "usr/lib", cp) == 0) {
+					if (strcmp(UCLIBC_RUNTIME_PREFIX UCLIBC_MULTILIB_DIR, cp) == 0
+					    || strcmp(UCLIBC_DEVEL_PREFIX UCLIBC_MULTILIB_DIR, cp) == 0) {
 						if (verbose >= 0)
 							warnx("You should remove `%s' from `%s'", cp, LDSO_CONF);
 						continue;
--- uClibc-0.9.33.2/utils/ldd.c.lib64~	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/utils/ldd.c	2013-01-07 02:09:28.238559787 +0100
@@ -128,7 +128,7 @@
 #endif
 
 #define ARRAY_SIZE(v)	(sizeof(v) / sizeof(*v))
-#define TRUSTED_LDSO	UCLIBC_RUNTIME_PREFIX "lib/" UCLIBC_LDSO
+#define TRUSTED_LDSO	UCLIBC_RUNTIME_PREFIX UCLIBC_MULTILIB_DIR "/" UCLIBC_LDSO
 
 struct library {
 	char *name;
@@ -460,9 +460,9 @@ static void locate_library_file(ElfW(Ehd
 
 	/* Lastly, search the standard list of paths for the library.
 	   This list must exactly match the list in uClibc/ldso/ldso/dl-elf.c */
-	path = UCLIBC_RUNTIME_PREFIX "lib:" UCLIBC_RUNTIME_PREFIX "usr/lib"
+	path = UCLIBC_RUNTIME_PREFIX UCLIBC_MULTILIB_DIR ":" UCLIBC_DEVEL_PREFIX UCLIBC_MULTILIB_DIR
 #ifndef __LDSO_CACHE_SUPPORT__
-	    ":" UCLIBC_RUNTIME_PREFIX "usr/X11R6/lib"
+	    ":" UCLIBC_RUNTIME_PREFIX "usr/X11R6/" UCLIBC_MULTILIB_DIR
 #endif
 	    ;
 	search_for_named_library(lib->name, buf, path);
--- uClibc-0.9.33.2/utils/Makefile.in.lib64~	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/utils/Makefile.in	2013-01-07 02:09:28.238559787 +0100
@@ -12,6 +12,8 @@ CFLAGS-utils := \
     $(SSP_ALL_CFLAGS) \
     -I$(top_srcdir)ldso/include \
     -DUCLIBC_RUNTIME_PREFIX=\"$(RUNTIME_PREFIX)\" \
+    -DUCLIBC_DEVEL_PREFIX=\"$(DEVEL_PREFIX)\" \
+    -DUCLIBC_MULTILIB_DIR=\"$(MULTILIB_DIR)\" \
     -DUCLIBC_LDSO=\"$(UCLIBC_LDSO)\" \
     -I$(top_srcdir)/$(KERNEL_HEADERS) \
     -DNOT_IN_libc \
--- uClibc-0.9.33.2/utils/porting.h.lib64~	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/utils/porting.h	2013-01-07 02:09:28.238559787 +0100
@@ -58,6 +58,10 @@
 # define UCLIBC_RUNTIME_PREFIX "/"
 #endif
 
+#ifndef UCLIBC_MULTILIB_DIR
+# define UCLIBC_MULTILIB_DIR "lib"
+#endif
+
 #undef UCLIBC_ENDIAN_HOST
 #define UCLIBC_ENDIAN_LITTLE 1234
 #define UCLIBC_ENDIAN_BIG    4321
