--- uClibc-0.9.33.3/include/fcntl.h~	2014-10-11 03:48:12.740470084 +0200
+++ uClibc-0.9.33.3/include/fcntl.h	2014-10-11 03:48:14.140497992 +0200
@@ -64,6 +64,11 @@ __BEGIN_DECLS
 # define AT_REMOVEDIR		0x200	/* Remove directory instead of
 					   unlinking file.  */
 # define AT_SYMLINK_FOLLOW	0x400	/* Follow symbolic links.  */
+# ifdef __USE_GNU
+#  define AT_NO_AUTOMOUNT	0x800	/* Suppress terminal automount
+					   traversal.  */
+#  define AT_EMPTY_PATH		0x1000	/* Allow empty relative pathname.  */
+# endif
 # define AT_EACCESS		0x200	/* Test access permitted for
 					   effective IDs, not real IDs.  */
 #endif
