--- uClibc-0.9.33.3/libc/sysdeps/linux/common/bits/socket.h.0034~	2014-09-23 09:54:35.000000000 +0200
+++ uClibc-0.9.33.3/libc/sysdeps/linux/common/bits/socket.h	2015-05-27 19:34:11.197999092 +0200
@@ -82,7 +82,8 @@ typedef __socklen_t socklen_t;
 #define PF_CAIF		37	/* CAIF sockets.  */
 #define PF_ALG		38	/* Algorithm sockets.  */
 #define PF_NFC		39	/* NFC sockets.  */
-#define	PF_MAX		40	/* For now..  */
+#define PF_VSOCK	40	/* vSockets.  */
+#define PF_MAX		41	/* For now..  */
 
 /* Address families.  */
 #define	AF_UNSPEC	PF_UNSPEC
@@ -126,6 +127,7 @@ typedef __socklen_t socklen_t;
 #define AF_CAIF		PF_CAIF
 #define AF_ALG		PF_ALG
 #define AF_NFC		PF_NFC
+#define AF_VSOCK	PF_VSOCK
 #define	AF_MAX		PF_MAX
 
 /* Socket level values.  Others are defined in the appropriate headers.
